docker.go

```diff
@@ -67,8 +67,64 @@ func StatusDockerContainer(name string) string {
 	return state[0]
 }
 
+// hasExitedDockerContainer checks if the container has exited.
+func hasExitedDockerContainer(name string) (bool, error) {
+	args := []string{"--format={{.State.Status}}", name}
+
+	stdout, _, exitCode := runDockerCommand("inspect", args...)
+
+	if exitCode != 0 || stdout == "" {
+		return false, fmt.Errorf("failed to run docker inspect command")
+	}
+
+	status := strings.TrimSpace(stdout)
+
+	if status == "exited" {
+		return true, nil
+	}
+
+	return false, nil
+}
+
 // ExitCodeDockerContainer returns the container exit code
-func ExitCodeDockerContainer(name string) (int, error) {
+func ExitCodeDockerContainer(name string, waitForExit bool) (int, error) {
+	// It makes no sense to try to retrieve the exit code of the container
+	// if it is still running. That's why this infinite loop takes care of
```

Comment:
nitpick: you could mention the timeout of the loop in the comment as well - as 'infinite loop' sounds dangerous otherwise :-)